### DISCUSSION TITLE
Where do I modify the trackball mouse cursor to go faster in default mode?

### DISCUSSION THREAD

**Discussion by KoichaDev** [Yowkees/keyball#586] [Q&A] [answered]
I am using the Keyball39 default whatever the trackball speed on the mouse is. I find this mouse movement from the trackball slow, so I have to press the `KB2` **(0x7E02)** 3 times to get the mouse cursor speed faster. I have to do this when unplugging my keyboard and plugging it back on my computer.

I know how to compile the Keyball39 firmware, but I am unsure where in the keyball39 code base file I need to modify to get the speed equal to the pressing 3 times of `KB2` **(0x7E02)** that will be my default mode. 

Can someone please guide me? 

Many thanks=

**Comment by CHBrianLiu** [comment, marked as answer]
hi! if i didn't get it wrong, the following line may be the configuration you're looking for. adding it to your `config.h` should work, at least for me.
```c
#define KEYBALL_CPI_DEFAULT 2000
```

**Reply by KoichaDev** [reply]
Thx for the reply! I will try it out later! What is the default number of the trackball is being used from the original firmware? Also, what is the number that is recommended to increment each time in order to either speed up the trackball?

**Reply by CHBrianLiu** [reply]
I think the default value is 500.
https://github.com/Yowkees/keyball/blob/b0cde713e9d63c4e5bb201df2fcfb1e45dc2d1fa/qmk_firmware/keyboards/keyball/lib/keyball/keyball.h#L23-L25

**Comment by KoichaDev** [comment]
> I think the default value is 500.
> 
> https://github.com/Yowkees/keyball/blob/b0cde713e9d63c4e5bb201df2fcfb1e45dc2d1fa/qmk_firmware/keyboards/keyball/lib/keyball/keyball.h#L23-L25

Fantastic! I had to change the value from `500` to `800`, and it worked perfectly fine! The only thing that is weird is that on my keyball right side when I flashed it to then I can see my LED shows `800`, but on the left side of my keyball, it still shows `500` even though I have tried to flash it.

Any idea why this is happening?

**Comment by CHBrianLiu** [comment]
Because I disabled the OLED feature to gain me more storage, I am afraid that I may not have too much to share regarding this issue. 🙂‍↔️
Let’s see if any one from the team is able to answer it!

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by KoichaDev** [reply]
Thank you for your help. I diagnosed my micro pro USB is broken after I flashed the new firmware. This is pretty strange, and never happened to me before.

I have to try and figure it out somehow
